src/fpm_sources.f90

```diff
@@ -17,6 +17,7 @@ module fpm_sources
 
 character(4), parameter :: fortran_suffixes(2) = [".f90", &
                                                   ".f  "]
+character(4), parameter :: c_suffixes(3) = [".c  ", ".h  ", ".cpp"]
```

Comment:
What about other C++ related suffixes? We don't have to support all of them, but `.hpp` seems important for headers at least.